client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,181 +70,167 @@ export default {
         scuttleDisabledExplanation = 'Your opponent has no point cards to scuttle';
       }
-      if (allMovesAreDisabled) {
-        scuttleDisabledExplanation = allMovesDisabledExplanation;
+      if (this.allMovesAreDisabled) {
+        scuttleDisabledExplanation = this.disabledText;
       }
-      const scuttleMove = {
+      return {
         displayName: 'Scuttle',
         eventName: 'scuttle',
         moveDescription: 'Scrap a lower point card',
-        disabled: scuttleDisabled || allMovesAreDisabled,
+        disabled: scuttleDisabled || this.allMovesAreDisabled,
         disabledExplanation: scuttleDisabledExplanation,
       };
+    },
+    oneOffMove() {
+      return {
+        displayName: 'One-Off',
+        eventName: 'oneOff',
+        moveDescription: this.selectedCard.ruleText,
+        disabled: this.allMovesAreDisabled,
+        disabledExplanation: this.disabledText,
+      };
+    },
+
+    targetedOneOffMove() {
+      let oneOffDisabled = !this.isPlayersTurn;
+      let oneOffDisabledExplanation = "It's not your turn";
+      if (this.isPlayersTurn) {
+        if (this.opponentQueenCount >= 2) {
+          oneOffDisabled = true;
+          oneOffDisabledExplanation = `You can't play a ${this.selectedCard.rank} while your opponent has two or more queens`;
+        } else {
+          let validTargetExists;
+          // Twos
+          if (this.selectedCard.rank === 2) {
+            const numOpFaceCards = this.$store.getters.opponent.faceCards.length;
+            const numOpJacks = this.$store.getters.opponent.points.reduce(
+              (jackCount, pointCard) => {
+                return jackCount + pointCard.attachments.length;
+              },
+              0
+            );
+            const numTotalTargets = numOpFaceCards + numOpJacks;
+            validTargetExists = numTotalTargets >= 1;
+            if (this.allMovesAreDisabled) {
+              oneOffDisabledExplanation = this.disabledText;
+            } else if (!validTargetExists) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no Royals to target';
+            }
+            // Nines
+          } else {
+            const numValidTargets =
+              this.$store.getters.opponent.points.length +
+              this.$store.getters.opponent.faceCards.length;
+            if (this.allMovesAreDisabled) {
+              oneOffDisabledExplanation = this.disabledText;
+            } else if (numValidTargets === 0) {
+              oneOffDisabled = true;
+              oneOffDisabledExplanation = 'There are no point cards or Royals to target';
+            }
+          }
+        }
+      }
+      return {
+        displayName: 'One-Off',
+        eventName: 'targetedOneOff',
+        moveDescription: this.selectedCard.ruleText,
+        disabled: oneOffDisabled || this.allMovesAreDisabled,
+        disabledExplanation: oneOffDisabledExplanation,
+      };
```

Comment:
Same idea as above. It's cleaner to have one variable for `oneOffDisabled` and for `oneOffDisabledExplanation` which get used in the return value down below, instead of having the variable and also comparing to the global stuff in the return statement. Basically it's funky in the current setup that we're referring to `this.isPlayersTurn` here but also implicitly down below when we use `this.allMovesAreDisabled` because that also reads from `this.isPlayersTurn` in the same way.

So instead, let's rejigger this to try to more efficiently leverage those existing variables and to reduce the number of times we need to check whether all moves are disabled. I'm writing this in the web editor in GH so I'm not 100% I got the format right, but see if you can rework this like so.
```suggestion
      let oneOffDisabled = this.allMovesAreDisabled;
      let oneOffDisabledExplanation = this.disabledText;
      if (!this.allMovesAreDisabled) {
        if (this.opponentQueenCount >= 2) {
          oneOffDisabled = true;
          oneOffDisabledExplanation = `You can't play a ${this.selectedCard.rank} while your opponent has two or more queens`;
        } else {
          let validTargetExists;
          // Twos
          if (this.selectedCard.rank === 2) {
            const numOpFaceCards = this.$store.getters.opponent.faceCards.length;
            const numOpJacks = this.$store.getters.opponent.points.reduce(
              (jackCount, pointCard) => {
                return jackCount + pointCard.attachments.length;
              },
              0
            );
            const numTotalTargets = numOpFaceCards + numOpJacks;
            validTargetExists = numTotalTargets >= 1;
            if (!validTargetExists) {
              oneOffDisabled = true;
              oneOffDisabledExplanation = 'There are no Royals to target';
            }
            // Nines
          } else {
            const numValidTargets =
              this.$store.getters.opponent.points.length +
              this.$store.getters.opponent.faceCards.length;
           if (numValidTargets === 0) {
              oneOffDisabled = true;
              oneOffDisabledExplanation = 'There are no point cards or Royals to target';
            }
          }
        }
      }
      return {
        displayName: 'One-Off',
        eventName: 'targetedOneOff',
        moveDescription: this.selectedCard.ruleText,
        disabled: oneOffDisabled || this.allMovesAreDisabled,
        disabledExplanation: oneOffDisabledExplanation,
      };
```